File: 3way/problem5/kidsWithCandies.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
bool* kidsWithCandies(int* candies, int candiesSize, int extraCandies, int* returnSize) {
    
    int max =  0;
    *returnSize = candiesSize;
    bool* result = (bool*)malloc(*returnSize * sizeof(bool));
    for(int i = 0; i < candiesSize; i++){
        max = candies[i] > max ? candies[i] : max;
    }

    for(int i = 0; i < candiesSize; i++){
        result[i] = candies[i] + extraCandies >= max;
    }

    return result;

}

File: 3way/problem5/kidsWithCandies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
    
    var result = []
    var max = 0;
    candies.forEach((candie, index) => {
        max = Math.max(max, candie);
    });

    for(let i = 0; i < candies.length; i++){
        result.push(candies[i] + extraCandies >= max);
    }

    return result;

};
